File: OneK-2/week_26/PGS_키패드 누르기.java
import java.util.*;

class Solution {
    public String solution(int[] numbers, String hand) {
        StringBuilder answer = new StringBuilder();
        int len = numbers.length;
        int leftLocation = 10;
        int rightLocation = 12;

        for (int i = 0; i < len; i++) {
            if (numbers[i] == 1 || numbers[i] == 4 || numbers[i] == 7) {
                answer.append("L");
                leftLocation = numbers[i];
            } else if (numbers[i] == 3 || numbers[i] == 6 || numbers[i] == 9) {
                answer.append("R");
                rightLocation = numbers[i];
            } else {
                int leftLen = Distance(leftLocation, numbers[i]);
                int rightLen = Distance(rightLocation, numbers[i]);
                if (leftLen > rightLen) {
                    answer.append("R");
                    rightLocation = numbers[i];
                } else if (leftLen < rightLen) {
                    answer.append("L");
                    leftLocation = numbers[i];
                } else {
                    if (hand.equals("left")) {
                        answer.append("L");
                        leftLocation = numbers[i];
                    } else {
                        answer.append("R");
                        rightLocation = numbers[i];
                    }
                }

            }

        }
        return answer.toString();
    }

    public int Distance(int location, int num) {
        if (location == 0) location = 11;
        if (num == 0) num = 11;
        int x = (location - 1) / 3;
        int y = (location - 1) % 3;
        int numX = num / 3;
        int numY = 1;

        return Math.abs(x - numX) + Math.abs(y - numY);
    }
}

File: hejoo345/week_26/PGS_키패드 누르기.js
function distance(cur, center) {
    return Math.abs(cur[0] - center[0]) + Math.abs(cur[1] - center[1]);
}

function solution(numbers, hand) {
    const keypad = {
        1: [0, 0], 2: [0, 1], 3: [0, 2],
        4: [1, 0], 5: [1, 1], 6: [1, 2],
        7: [2, 0], 8: [2, 1], 9: [2, 2],
        10: [3, 0], 0: [3, 1], 12: [3, 2]
    };
    let cur_left = [3, 0];
    let cur_right = [3, 2];
    let result = '';

    numbers.forEach(num => {
        let cur_num = keypad[num];
        if (cur_num[1] === 0) {
            cur_left = cur_num;
            result += 'L';
        } else if (cur_num[1] === 2) {
            cur_right = cur_num;
            result += 'R';
        } else {
            let left_dis = distance(cur_left, cur_num);
            let right_dis = distance(cur_right, cur_num);
            if (left_dis === right_dis) {
                if (hand === 'left') {
                    cur_left = cur_num;
                    result += 'L';
                } else {
                    cur_right = cur_num;
                    result += 'R';
                }
            } else if (left_dis < right_dis) {
                cur_left = cur_num;
                result += 'L';
            } else {
                cur_right = cur_num;
                result += 'R';
            }
        }
    });
    return result;
}